src/checkout-buttons/strategies/masterpass-button-strategy.ts

```diff
@@ -0,0 +1,144 @@
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                this._checkout = stateCheckout.checkout.getCheckout();
+                if (!this._checkout || !this._checkout.cart.id) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                return this._masterpassScriptLoader.load(this._paymentMethod.config.testMode)
+                    .then(masterpass => {
+                        this._masterpassClient = masterpass;
+                        this._createSignInButton(masterpassOptions);
+                    });
+        })
+        .then(() => super.initialize(options));
+    }
+
+    deinitialize(options: CheckoutButtonOptions): Promise<void> {
+        if (!this._isInitialized) {
+            return super.deinitialize(options);
+        }
+
+        this._paymentMethod = undefined;
+        this._checkout = undefined;
+
+        if (this._signInButton && this._signInButton.parentNode) {
+            this._signInButton.removeEventListener('click', this._handleWalletButtonClick);
+            this._signInButton.parentNode.removeChild(this._signInButton);
+            this._signInButton = undefined;
+        }
+
+        return super.deinitialize(options);
+    }
+
+    private _createSignInButton(masterpassOptions: MasterpassButtonInitializeOptions): void {
+        const { container } = masterpassOptions;
+        const buttoncontainer = document.querySelector(`#${container}`);
+
+        if (!buttoncontainer) {
+            throw new Error('Need a container to place the button');
+        }
+
+        const button = document.createElement('input');
+
+        button.type = 'image';
+        button.src = 'https://static.masterpass.com/dyn/img/btn/global/mp_chk_btn_160x037px.svg';
+        buttoncontainer.appendChild(button);
+        this._signInButton = button;
+        this._signInButton.addEventListener('click', this._handleWalletButtonClick);
+    }
+
+    private _createMasterpassPayload(): MasterpassCheckoutOptions {
+        return {
+            checkoutId: this.paymentMethod.initializationData.checkoutId,
+            allowedCardTypes: this.paymentMethod.initializationData.allowedCardTypes,
+            amount: this.checkout.cart.cartAmount.toString(),
+            currency: this.checkout.cart.currency.code,
+            cartId: this.checkout.cart.id,
+            suppressShippingAddress: true,
+        };
+    }
+
+    @bind
+    private _handleWalletButtonClick(): void  {
+        const payload = this._createMasterpassPayload();
+        this.masterpassClient.checkout(payload);
+    }
+
+    private get masterpassClient(): Masterpass {
+        if (!this._masterpassClient) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._masterpassClient;
+    }
+
+    private get paymentMethod(): PaymentMethod {
```

Comment:
Please see my previous comment regarding the use of `get` and the name of the method. If you do want to have a private method for retrieving the `PaymentMethod` object, you should retrieve the data directly from `store` instead. This is because keeping a separate reference is harder to maintain as you always have to remember to update the reference when the object gets stale.

```ts
private _getPaymentMethod(): PaymentMethod {
    const paymentMethod = state.paymentMethods.getPaymentMethod(this._methodId);

    if (!paymentMethod) {
        throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
    }

    return paymentMethod;
}
```

That'd also simply your `initialize` method.